module/builder/consensus/builder.go

```diff
@@ -383,9 +383,10 @@ func (b *Builder) getInsertableSeals(parentID flow.Identifier) ([]*flow.Seal, er
 	//    Therefore, we only have to inspect the results incorporated in unsealed blocks.
 	sealsSuperset := make(map[uint64][]*flow.IncorporatedResultSeal) // map: executedBlock.Height -> candidate Seals
 	sealCollector := func(header *flow.Header) error {
-		block, err := b.blocks.ByID(header.ID())
+		blockID := header.ID()
+		block, err := b.blocks.ByID(blockID)
```

Comment:
[optional suggestion for performance improvements]
using `b.index` and the retrieving the results individually is more efficient that assembling the entire block payload